common/src/app/common/logic/variant_properties.cljc

```diff
@@ -117,86 +117,81 @@
-                                get-base-name
-                                cfh/split-path
-                                count)
-
-        max-path-items     (apply max (map calc-num-props shapes))
+(defn generate-make-shapes-variant
+  [changes shapes variant-container]
+  (let [data           (pcb/get-library-data changes)
+        objects        (pcb/get-objects changes)
+        variant-id     (:id variant-container)
 
         ;; If we are cut-pasting a variant-container, this will be null
         ;; because it hasn't any shapes yet
-        first-comp-id      (->> variant-container
-                                :shapes
-                                first
-                                (get objects)
-                                :component-id)
-
-        variant-properties (get-in data [:components first-comp-id :variant-properties])
-        num-props          (count variant-properties)
-        num-new-props      (if (or (nil? first-comp-id)
-                                   (< max-path-items num-props))
-                             0
-                             (- max-path-items num-props))
-        total-props        (+ num-props num-new-props)
-
-        changes            (nth
-                            (iterate #(generate-add-new-property % (:id variant-container)) changes)
-                            num-new-props)
-
-        changes            (pcb/update-shapes changes (map :id shapes)
-                                              #(assoc % :variant-id (:id variant-container)
-                                                      :name (:name variant-container)))]
+        first-comp-id  (->> variant-container
+                            :shapes
+                            first
+                            (get objects)
+                            :component-id)
+
+        base-props     (->> (get-in data [:components first-comp-id :variant-properties])
+                            (map #(assoc % :value "")))
+        num-base-props (count base-props)
+
+        [cpath cname]  (cfh/parse-path-name (:name variant-container))
+        container-name (:name variant-container)
+
+        generate-new-properties
+        (fn [shape min-props]
+          (if (ctk/is-variant? shape)
+            (generate-new-properties-from-variant shape min-props data container-name base-props)
+            (generate-new-properties-from-non-variant shape min-props container-name base-props)))
+
+        total-props    (apply max (map #(-> (generate-new-properties % num-base-props) count)
+                                       shapes))
+
+        num-new-props  (if (or (zero? num-base-props)
+                               (< total-props num-base-props))
+                         0
+                         (- total-props num-base-props))
+
+        changes        (nth
+                        (iterate #(generate-add-new-property % variant-id) changes)
+                        num-new-props)
```

Comment:
`nth` on sequence will iterate over all sequence until the index. This is just information, I don't know the algorithm behind this code so just comment it to you to be aware of it.